### PR TITLE
Quasar 17.7.0 foun 155

### PR DESCRIPTION
**Pull request by LihiAvsh**

### CODE REVIEW

File: infra/cache/kInfraRedisCacheWrapper.php [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+			return $this->redis->get($key);
+		}, array($key));
+	}
+
+	/* (non-PHPdoc)
+     * @see kBaseCacheWrapper::set()
+     */
+	protected function doSet ($key, $var, $expiry = 0)
+	{
+		return $this->callAndDetectErrors(function($key, $var, $expiry = 0) {
```
Comment (yossipapi) [inline infra/cache/kInfraRedisCacheWrapper.php:97]: NEed to handle the case where expiry is 0 which is different than how Memcache implements it.

**Comment by kaltura-hooks** [comment]
Hi @LihiAvsh,
Thank you for contributing this pull request!
Please sign the Kaltura CLA so we can review and merge your contribution.
Learn more at http://bit.ly/KalturaContrib

File: infra/cache/kInfraRedisCacheWrapper.php [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+
+				self::safeLog("got timeout error while connecting to redis...");
+			}
+		}
+
+		else
+		{
+			try
+			{
+				if ($this->persistent)
```
Comment (yossipapi) [inline infra/cache/kInfraRedisCacheWrapper.php:287]: The creation does not match method signature:
redis_cluster_init(redisCluster *c, HashTable *ht_seeds, double timeout,
                               double read_timeout, int persistent, zend_string *user,
                               zend_string *pass, zval *context)
                               
Also better consolidate to one while sending the persistent value as param.

File: infra/cache/kInfraRedisCacheWrapper.php [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+		}
+		$this->redis = null;
+	}
+
+	/* (non-PHPdoc)
+     * @see kBaseCacheWrapper::get()
+     */
+	protected function doGet ($key)
+	{
+		return $this->callAndDetectErrors('get',array($key));
```
Comment (yossipapi) [inline infra/cache/kInfraRedisCacheWrapper.php:76]: Add spacing after the , 
For example
$this->callAndDetectErrors('get', array($key));

File: infra/cache/kInfraRedisCacheWrapper.php [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+		$this->close();
+
+		if ($this->connectAttempts >= self::MAX_CONNECT_ATTEMPTS)
+		{
+			return false;
+		}
+
+		$connectResult = false;
+		$connStart = microtime(true);
+		if(!$this->cluster)
```
Comment (yossipapi) [inline infra/cache/kInfraRedisCacheWrapper.php:257]: Try and consolidate the code that handles connections to cluster mode and none cluster mode.

**Pull request merged**